### ISSUE TITLE
Progress is reported incorrectly for Texture Extraction build task

### ISSUE THREAD

**Issue by focustense** [focustense/easymod#165]
Several users report apparent freezes during this step. For the majority of these users, it eventually recovers and finishes the build after anywhere from 10 seconds to about a minute - this is repro in local testing. For a small minority of users, it seems to never recover, and gets stuck for hours if not permanently.

The proximate cause of the former issue appears to be spamming tasks all at once (each NIF undergoing extraction gets its own task, this is by design), instead of waiting for previous tasks to finish before starting new ones. This makes the build progress UI think it's close to 100%, but in reality there are just dozens or hundreds of background tasks in a not-started state. No additional progress is made until they all complete.

This **may be** (but is by no means guaranteed to be) the root cause of the freezing experienced by the latter minority of users. It is possible that having all of those tasks pending is creating some kind of deadlock, thread pool exhaustion, I/O thrashing, triggering Windows Defender protection, etc... unclear exactly how it might be linked but it certainly can't be helping. The issue needs to be fixed regardless, and there's a chance that fixing it will also resolve the rare freezing; if not, having an accurate progress display will still help those users identify if there's a specific file causing it to fail.
